miden-lib/asm/miden/kernel_invocation.masm

```diff
@@ -0,0 +1,374 @@
+# KERNEL PROCEDURE OFFSETS
+# =================================================================================================
+
+# OFFSET CONSTANTS
+# -------------------------------------------------------------------------------------------------
+
+const.GET_ACCOUNT_ID_OFFSET=0
+const.GET_ACCOUNT_NONCE_OFFSET=1
+const.GET_INITIAL_ACCOUNT_HASH_OFFSET=2
+const.GET_CURRENT_ACCOUNT_HASH_OFFSET=3
+const.INCR_ACCOUNT_NONCE_OFFSET=4
+const.GET_ACCOUNT_ITEM_OFFSET=5
+const.SET_ACCOUNT_ITEM_OFFSET=6
+const.GET_ACCOUNT_MAP_ITEM_OFFSET=7
+const.SET_ACCOUNT_MAP_ITEM_OFFSET=8
+const.SET_ACCOUNT_CODE_OFFSET=9
+const.ACCOUNT_VAULT_GET_BALANCE_OFFSET=10
+const.ACCOUNT_VAULT_HAS_NON_FUNGIBLE_ASSET_OFFSET=11
+const.ACCOUNT_VAULT_ADD_ASSET_OFFSET=12
+const.ACCOUNT_VAULT_REMOVE_ASSET_OFFSET=13
+const.GET_NOTE_ASSETS_INFO_OFFSET=14
+const.GET_NOTE_INPUTS_HASH_OFFSET=15
+const.GET_NOTE_SENDER_OFFSET=16
+const.GET_BLOCK_NUMBER_OFFSET=17
+const.GET_BLOCK_HASH_OFFSET=18
+const.GET_INPUT_NOTES_COMMITMENT_OFFSET=19
+const.GET_OUTPUT_NOTES_HASH_OFFSET=20
+const.CREATE_NOTE_OFFSET=21
+const.ADD_ASSET_TO_NOTE_OFFSET=22
+const.GET_ACCOUNT_VAULT_COMMITMENT_OFFSET=23
+const.MINT_ASSET_OFFSET=24
+const.BURN_ASSET_OFFSET=25
+const.GET_FUNGIBLE_FAUCET_TOTAL_ISSUANCE_OFFSET=26
+const.GET_NOTE_SERIAL_NUMBER_OFFSET=27
+const.EXEC_KERNEL_PROC_OFFSET=28
```

Comment:
Let's group these procedures by their respective categories. These are:

- `current_account`
- `current_faucet`
- `current_note`
- `transaction`

Also, why is `EXEC_KERNEL_PROC_OFFSET` here? I think we should remove it.